### PR TITLE
Adaptive sdxl

### PR DESCRIPTION
**Pull request by satsumas**
Adds parameters used to enable T2I requests.

To test:

1. Check out the following repositories at these branches:

generator_server `adaptive_sdxl` 

stable-diffusion,` ENG-811-PLATFORM-408-hatch` branch

Interfaces, branch `adaptive_sdxl`

2. Get the following SDXL 2.2.2 T2I adapters from the server at `cw.hpc.stability.ai:/mnt/nvme/home/nitrosocke/t2i-adapter-trainer/training_logs/`, and put them in `models/t2iadapters/models`

`sdxl-sketch_e0_gs16000.pth`

`sdxl-canny_e0_gs10000.pth`

`sdxl-depth_e1_gs24000.pth`

3. Get the example images and the 3rd party models by running the scripts as instructed in stable_diffusion/src/stablecore/modules/t2iadapter/README.md.

4. Then, from inside `generator_server/` run the following. Note that the `MODELS_DIR` is specific to this advanced inference use case:
```
hatch -e sdxl shell
pip install -e src/generator_server/lib/stable_diffusion/
pip install -e .
pip install -e /data/workspace/sandbox/stability-sdk/ #ENG-811-PLATFORM-408-hatch branch
export MODELS_DIR="/path/to/generator_server/deploy/models/stable-diffusion-xl-beta-v2-2-2-advanced"
export $(grep -v '^#' $MODELS_DIR/model.env | xargs -d '\n')


ADAPTER_PATH="/path/to/generator_server/src/generator_server/lib/stable_diffusion/models"  # inside here are your adapters and 3rd party models 
SD_REPO_PATH="/path/to/stable-diffusion-staging" ONEFLOW_PATH="/path/to/generator_server/src/generator_server/lib/stable_diffusion/outputs/txt2img-samples/serialized" python src/generator_server/grpc_server.py
```

Now the server is running, you can send T2I requests using the following code:

```
import io
import os
import warnings

from PIL import Image
from stability_sdk import client
import stability_sdk.interfaces.gooseai.generation.generation_pb2 as generation

# Our Host URL should not be prepended with "https" nor should it have a trailing slash.
os.environ['STABILITY_HOST'] = 'localhost:50051'

# Sign up for an account at the following link to get an API Key.
# https://dreamstudio.ai/

# Click on the following link once you have created an account to be taken to your API Key.
# https://dreamstudio.ai/account

# Paste your API Key below.

os.environ['STABILITY_KEY'] = 'anything-will-do'
# Set up our connection to the API.
stability_api = client.StabilityInference(
    key=os.environ['STABILITY_KEY'], # API Key reference.
    host=os.environ['STABILITY_HOST'], # Host reference.
    verbose=True, # Print debug messages.
    engine="stable-diffusion-xl-beta-v2-2-2", # Set the engine to use for generation. For SD 2.0 use "stable-diffusion-v2-0".
    # Available engines: stable-diffusion-v1 stable-diffusion-v1-5 stable-diffusion-512-v2-0 stable-diffusion-768-v2-0
    # stable-diffusion-512-v2-1 stable-diffusion-768-v2-1 stable-diffusion-xl-beta-v2-2-2 stable-inpainting-v1-0 stable-inpainting-512-v2-0
)


# Set up our initial generation parameters.
answers = stability_api.generate(
    prompt= "A car with wings",
    seed=82675780,
    adapter_type = generation.SKETCH,
    adapter_strength = 0.75,
    adapter_init_type = generation.ADAPTER_IMAGE,
    sampler=generation.SAMPLER_K_DPMPP_SDE,
    init_image = Image.open("path/to/generator_server/src/generator_server/lib/stable_diffusion/examples/sketch/car.png")
)

# Set up our warning to print to the console if the adult content classifier is tripped.
# If adult content classifier is not tripped, display generated image.
for resp in answers:

    for artifact in resp.artifacts:
        if artifact.finish_reason == generation.FILTER:
            warnings.warn(
                "Your request activated the API's safety filters and could not be processed."
                "Please modify the prompt and try again.")
        if artifact.type == generation.ARTIFACT_IMAGE:
            global img
            img = Image.open(io.BytesIO(artifact.binary))
            img.save(str(artifact.seed)+ ".png") # Save our generated images its seed number as the filename.
            print(f"Generated image saved as {artifact.seed}.png")

```

### CODE REVIEW

File: .github/workflows/publish-to-pypi.yml
Comment (satsumas) [inline .github/workflows/publish-to-pypi.yml:1]: TODO: match `main`

File: .github/workflows/unit_testing.yaml [hunk 1]
```diff
@@ -25,8 +25,8 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
-      - name: Install package
+          python -m pip install --upgrade hatch
```
Comment (satsumas) [inline .github/workflows/unit_testing.yaml:28]: TODO revert changes to match `main`

File: pyproject.toml [hunk 2]
```diff
@@ -1,7 +1,119 @@
 [build-system]
-requires = ["setuptools"]
-build-backend = "setuptools.build_meta"
+requires = ["hatchling"]
```
Comment (satsumas) [inline pyproject.toml:2]: TODO: revert

File: src/stability_sdk/__init__.py [hunk 3]
```diff
@@ -1,3 +1,4 @@
+__version__ = "0.8.0"
```
Comment (satsumas) [inline src/stability_sdk/__init__.py:1]: TODO: revert

File: src/stability_sdk_sagemaker/README.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
+# Stability AI Sagemaker SDK
```
Comment (satsumas) [inline src/stability_sdk_sagemaker/README.md:1]: TODO: remove

File: src/stability_sdk_sagemaker/models.py [hunk 5]
```diff
@@ -0,0 +1,7 @@
+from boto3.session import Session
+
```
Comment (satsumas) [inline src/stability_sdk_sagemaker/models.py:2]: TODO: remove

File: src/stability_sdk_sagemaker/predictor.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import json
```
Comment (satsumas) [inline src/stability_sdk_sagemaker/predictor.py:1]: TODO: remove

File: pyproject.toml [hunk 7]
```diff
@@ -1,7 +1,3 @@
 [build-system]
 requires = ["setuptools"]
-build-backend = "setuptools.build_meta"
-
-[tool.pytest.ini_options]
-log_cli = true
-log_cli_level = "DEBUG"
```
Comment (pharmapsychotic) [inline pyproject.toml:7]: We probably still want this for the pytest's that run with Github actions yeah?

**Review by pharmapsychotic — APPROVED** [review]

**Pull request merged**